File: JavaApp/src/main/java/com/charlezz/javaapp/feature/base/BaseViewHolder.java
package com.charlezz.javaapp.feature.base;

import androidx.databinding.ViewDataBinding;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class BaseViewHolder<VDB extends ViewDataBinding> extends RecyclerView.ViewHolder {
    private VDB binding;
    public BaseViewHolder(@NonNull VDB binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public VDB getBinding() {
        return binding;
    }
}

File: KotlinApp/src/main/java/com/charlezz/finalarchitecture/feature/base/BaseViewHolder.kt
package com.charlezz.finalarchitecture.feature.base

import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView

abstract class BaseViewHolder<VDB : ViewDataBinding>(val binding: VDB)
    : androidx.recyclerview.widget.RecyclerView.ViewHolder(binding.root)